source/Octopus.Server.Client/Tasks/AsyncLazy.cs

```diff
@@ -0,0 +1,42 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Octopus.Client.Tasks
+{
+    /// <summary>
+    /// The <see cref="Lazy{T}"/> implementation with <see cref="CancellationToken"/> support.
+    /// </summary>
+    public class AsyncLazy<T> where T : class
```

Comment:
This is an import from OctopusDeploy.

Reply:
Did this have any tests in `OctopusDeploy` that should come across also?

Reply:
Let me find out, thanks for the reminder!

Reply:
There is no **specific** unit test. We have tests around the owner class that will exercise the access on `AsyncLazy` variable. This is similar to IE: `ShouldCreateNewMachine` test here, which will eventually exercise the usage.